saleor/graphql/account/mutations/account.py

```diff
@@ -33,7 +33,7 @@ class AccountRegisterInput(graphene.InputObjectType):
         description=(
             "Base of frontend URL that will be needed to create confirmation URL."
         ),
-        required=True,
+        required=settings.ENABLE_ACCOUNT_CONFIRMATION_BY_EMAIL,
```

Comment:
I think schema.graphql should be always static, e.g. changing env variable `ENABLE_ACCOUNT_CONFIRMATION_BY_EMAIL` shouldn't require re-generating it. Maybe we could just change it to `required=False` and have a validation rule in `clean_input` of that mutation to raise an error if this setting is enabled and the value was not provided.

Reply:
Now I see that we have changes in `clean_input` as well - maybe it's enough; if `validate_storefront_url` returns a meaningful error when url is not provided then it's fine.